legend-engine-language-pure-compiler/src/test/java/org/finos/legend/engine/language/pure/compiler/test/fromGrammar/TestDataCompilationFromGrammar.java

```diff
@@ -264,13 +273,19 @@ public void testModelStoreDataCompilation()
         Assert.assertTrue(dataElement1._data() instanceof Root_meta_pure_data_ModelStoreData);
         Root_meta_pure_data_ModelStoreData modelStoreData1 = (Root_meta_pure_data_ModelStoreData) dataElement1._data();
 
+        Assert.assertEquals(2, modelStoreData1._modelData().size());
         PackageableElement cl1 = result1.getTwo().getPackageableElement("my::Person");
-        InstanceValue collection1 = (InstanceValue) modelStoreData1._instances().getMap().get(cl1);
+
+        Root_meta_pure_data_ModelInstanceData modelInstanceData1 = (Root_meta_pure_data_ModelInstanceData) modelStoreData1._modelData()
+            .select(e -> e instanceof Root_meta_pure_data_ModelInstanceData && e._model() == cl1).getOnly();
+        InstanceValue collection1 = modelInstanceData1._instances();
         Assert.assertEquals(2, collection1._values().size());
         Assert.assertTrue(collection1._values().allSatisfy(SimpleFunctionExpression.class::isInstance));
 
         PackageableElement cl2 = result1.getTwo().getPackageableElement("my::Address");
-        InstanceValue collection2 = (InstanceValue) modelStoreData1._instances().getMap().get(cl2);
+        Root_meta_pure_data_ModelInstanceData modelInstanceData2 = (Root_meta_pure_data_ModelInstanceData) modelStoreData1._modelData()
```

Comment:
Where are we adding compilation tests for new test suite?

Reply:
was leaving testing for the end. updated the tests